phoenix-core/src/main/java/org/apache/phoenix/schema/tool/SchemaExtractionProcessor.java

```diff
@@ -502,25 +504,84 @@ private String getColumnInfoStringForTable(PTable table) {
         return getColumnInfoString(table, colInfo, columns, pkColumns);
     }
 
-    private boolean shouldContainQualifier(PTable table, List<PColumn> columns)
+    private boolean hasEncodedQualifier(PTable table)
     {
-        if (columns.size() == 0 || shouldGenerateWithDefaults) {
-            return false;
-        }
+        return table.getColumns().size() > 0
+                && !shouldGenerateWithDefaults
+                && table.getType() == PTableType.TABLE
+                && table.getEncodingScheme() != PTable.QualifierEncodingScheme.NON_ENCODED_QUALIFIERS;
+    }
 
-        return table.getType() == PTableType.TABLE && table.getEncodingScheme() != PTable.QualifierEncodingScheme.NON_ENCODED_QUALIFIERS;
+    private boolean isConsecutive(List<Integer> list) {
+        for (int i = 1; i < list.size(); i++) {
+            if (list.get(i - 1) + 1 != list.get(i)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private List<String> getNonConsecutiveQualifierCounterFamilies(PTable table) {
```

Comment:
I'd prefer simply getNonConsecutiveQualifierFamilies

Reply:
Also for the variable names, use something like idValues, or encodedIds.
Try to avoid the word "counter" when not referring to the actual counter fields in syscat.